### ISSUE TITLE
tf.matmul vs K.Dot

### ISSUE THREAD

**Issue by lschaupp** [keras-team/keras#9779]
Quick question: (tensorflow 1.4/Keras 2.2)

Is there any difference between tf.matmul and keras dot function?
Seems to me that the dot function needs a specific axis, while the matmul function only
needs the two matrices. Is there any other difference? Any examples?

Thanks!

**Comment by ibealec** [comment]
They actually both work without specifying an axis. TensorFlow uses float64 by default whereas Keras uses float32. Using `np.random.seed(42)` and `r = np.random.rand(4, 5)`,
``` 
x = tf.constant(r)  
cov = tf.matmul(x, tf.transpose(x))
```
returns `Tensor("MatMul:0", shape=(4, 4), dtype=float64)`
```
x = K.constant(x_val)
cov = K.dot(x, K.transpose(x))
```
returns `Tensor("MatMul_1:0", shape=(4, 4), dtype=float32)`

Note that there is a variance of about 1e-7 between the two, as pointed out in [https://stackoverflow.com/a/44100246/6647539](url)

**Comment by lschaupp** [comment]
thanks for the info! Does it count also for 2D tensors which are not having symmetrical dimensions?
(e.g. 3x4 =A, 2x3=B)

**Comment by zekedran** [comment]
Higher dimensions? 

```
a.shape          # (125, 125, 1, 3)
b/shape         # (125, 125, 3, 1)
np.matmul(a,b).shape        # (125,125,1,1)
K.dot(ab).shape                  # (125, 125, 1, 125, 125, 1)
```

**Comment by zekedran** [comment]
I ended up using `tf.matmul` importing tf
It doesn't matter, we are doing tensor operations and it's completely fine

**Comment by hummosa** [comment]
I'm facing the same issue as @zekedran higher dimensions are a problem in Keras. Clarifying proper use would be appreciated.
